### PR TITLE
bug fixes to add 4 more azure_svc crates 

### PR DESCRIPTION
**Pull request by ctaggart**
- add azure_svc_deviceupdate
- add azure_svc_digitaltwins
- add azure_svc_servicefabric
- add azure_svc_marketplacecatalog

2 more just need boxing now: 

- machinelearningservice
- purview

Changes:

- fix parsing of openapi specs with form data. support for form data not yet #500
- fix some idents with rust keywords or colons

### CODE REVIEW

File: services/autorust/codegen/src/codegen_operations.rs [hunk 1]
```diff
@@ -281,18 +282,20 @@ fn create_function(cg: &CodeGen, operation: &WebOperation) -> Result<TokenStream
...
-                    });
-                } else {
-                    ts_request_builder.extend(quote! {
-                        if let Some(#param_name_var) = #param_name_var {
-                            req_builder = req_builder.form(#param_name_var);
-                        }
-                    });
-                }
+            ParameterType::FormData => {
+                // form data not yet supported
```
Comment (bmc-msft) [inline services/autorust/codegen/src/codegen_operations.rs:286]:
```suggestion
                unimplemented!("form data not yet supported");
```

File: services/autorust/codegen/src/identifier.rs [hunk 2]
```diff
@@ -100,6 +119,7 @@ fn is_keyword(word: &str) -> bool {
         "abstract"
             | "alignof"
             | "as"
+            | "async"
```
Comment (bmc-msft) [inline services/autorust/codegen/src/identifier.rs:122]: 👍

File: services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs [hunk 3]
```diff
@@ -166,7 +166,7 @@ pub mod adds_services {
     use super::{models, API_VERSION};
     pub async fn list(
         operation_config: &crate::OperationConfig,
-        filter: Option<&str>,
+        u24filter: Option<&str>,
```
Comment (bmc-msft) [inline services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs:169]: Where did the u24 come from?

**Review by bmc-msft — COMMENTED** [review]
In general, this is positive.  I'm not sure why a number of generated crates have some function arguments prepended with u24.   Why was this included?

File: services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs (hunk 3 above)
Comment (cataggar) [reply services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs:169]: `$filter` becomes `u24filter`. The `u24` represents the `$`. http://www.unicode-symbol.com/u/0024.html

Comment (cataggar) [reply services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs:169]: Details are in https://github.com/Azure/azure-sdk-for-rust/blob/8ea32f9193fb468686b869247fd4aa0bc799197a/services/autorust/codegen/src/identifier.rs#L65-L85

Comment (bmc-msft) [reply services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs:169]: Thanks for the clarification on how this is generated.

That is not obvious from the perspective of users of the library.  Probably just as important though, it's not how other SDKs solve this.  

Ref:

1. Azure SDK For Go:
https://github.com/Azure/azure-sdk-for-go/blob/b63dc6cf8623c3e970b8e2ee5a16f948f3a96636/services/batch/2020-09-01.12.0/batch/jobschedule.go#L866-L879
2. Azure SDK for Python:
https://github.com/Azure/azure-sdk-for-python/blob/596e2de8cd39d1de34e0715ef1f8c47c93c4bf31/sdk/batch/azure-batch/azure/batch/operations/_job_schedule_operations.py#L1007-L1010
3. Azure SDK for Net: https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/batch/Microsoft.Azure.Batch/src/GeneratedProtocol/Models/JobListPreparationAndReleaseTaskStatusOptions.cs#L37-L54

File: services/mgmt/cdn/src/package_2017_10/models.rs [hunk 4]
```diff
@@ -156,15 +156,15 @@ pub struct DeliveryRuleUrlPathCondition {
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct UrlPathConditionParameters {
     #[serde(rename = "@odata.type")]
-    pub odata_type: url_path_condition_parameters::Odata_type,
+    pub odata_type: url_path_condition_parameters::OdataType,
```
Comment (cataggar) [inline services/mgmt/cdn/src/package_2017_10/models.rs:159]: 👍

File: services/mgmt/appplatform/src/package_preview_2021_09/models.rs [hunk 5]
```diff
@@ -997,7 +997,7 @@ pub mod supported_runtime_version {
     pub enum Platform {
         Java,
         #[serde(rename = ".NET Core")]
-        U2eNetCore,
+        NetCore,
```
Comment (cataggar) [inline services/mgmt/appplatform/src/package_preview_2021_09/models.rs:1000]: 👍

File: services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs (hunk 3 above)
Comment (cataggar) [reply services/mgmt/adhybridhealthservice/src/package_2014_01/operations.rs]: Thanks for the links. I agree. I made the necessary adjustments and added more Identifier unit tests. It is tricky.

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**